### PR TITLE
correctly libcurl recipe to use self.run("....", run_environment=True)

### LINKED ISSUE #991: [package] libcurl/all: output misleading
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libcurl/all**
  * Operating System+version: **not Windows**
  * Compiler+version: **not Visual Studio**

Implementation of `build()` method for not-Visual Studio contains some output that can lead to misunderstandings:

```python
    def _build_with_autotools(self):
        env_run = RunEnvironment(self)
        # run configure with *LD_LIBRARY_PATH env vars
        # it allows to pick up shared openssl
        self.output.info("Run vars: " + repr(env_run.vars))
```

Conan applies the environment from the profile before entering the `build` function, so the `Run vars:...` message could be improved if it says all the output (see issue https://github.com/conan-io/conan/issues/6586).

Or, even better, we could move this recipe to use `self.run("....", run_environment=True)` which applies the `RunEnvironment` too.

### PR DESCRIPTION
**Pull request by prince-chrismc**
fixes #991

Specify library name and version:  **libcurl/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
The changes I made only affect Linux and mac, 

I've tested locally
```
libcurl/7.64.1: Exported revision: cb3e4b18c29d499ad62493629fbff38c
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=15
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
```

And can not reproduce the problem

**Comment by madebr** [comment]
@prince-chrismc 
I agree, there is some problem on CI.
In the last log, the test `Performing Curl Test TIME_WITH_SYS_TIME` failed because it could not find the test source which cmake just created.
The builder probably ran out of disk space or something similar.
@uilianries

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -262,25 +262,20 @@ def _patch_mingw_files(self):
                 tools.save(os.path.join('lib', 'Makefile.am'), added_content, append=True)
 
     def _build_with_autotools(self):
-        env_run = RunEnvironment(self)
-        # run configure with *LD_LIBRARY_PATH env vars
-        # it allows to pick up shared openssl
-        self.output.info("Run vars: " + repr(env_run.vars))
-        with tools.environment_append(env_run.vars):
```
Comment (madebr) [inline recipes/libcurl/all/conanfile.py:269]: By removing `RunEnvironment`, does this recipe still build with shared `ssh2`/`openssl`/`nghttp2`/`zlib`?

Comment (prince-chrismc) [reply recipes/libcurl/all/conanfile.py:269]: it is still present, just using a newer syntax `self.run('./buildconf', win_bash=use_win_bash, run_environment=True)`

Comment (madebr) [reply recipes/libcurl/all/conanfile.py:269]: No, `./buildconf` runs `autorconf` (which regenerates the `configure` script).
The `run_environment` is required in `autotools.configure()`.

Comment (prince-chrismc) [reply recipes/libcurl/all/conanfile.py:269]: Finally had the chance to test it locally `conan create . -o openssl:shared=True` passes and prints the correct Openssl version

Comment (uilianries) [reply recipes/libcurl/all/conanfile.py:269]: Could you please shared the output?

**Comment by uilianries** [comment]
> 
> 
> @prince-chrismc
> I agree, there is some problem on CI.
> In the last log, the test `Performing Curl Test TIME_WITH_SYS_TIME` failed because it could not find the test source which cmake just created.
> The builder probably ran out of disk space or something similar.
> @uilianries

I think I saw it in the past

File: recipes/libcurl/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/libcurl/all/conanfile.py:269]: https://gist.github.com/prince-chrismc/e1247a4f3c13dcecc88ca59e0f656769

https://github.com/openssl/openssl/releases/tag/OpenSSL_1_1_1 version I have installed, which is not the one reported by the logs

From my understanding this is was we are expecting

**Comment by prince-chrismc** [comment]
@uilianries I forced pushed and missed you review, I appologize for deleting you comments

File: recipes/libcurl/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/libcurl/all/conanfile.py:275]: > No, `./buildconf` runs `autoreconf` (which regenerates the `configure` script).
> The `run_environment` is required in `autotools.configure()`.

Turns out this was true for clang =)

**Pull request merged**